database/database.go

```diff
@@ -114,46 +125,29 @@ func Initialize(path string) (err error) {
 	return
 }
 
-// NewDatabase creates a new database object. The database directory is assumed
-// to be already initialized.
-func NewDatabase(
-	path string,
-	readonly bool) (d *Database, err error) {
-
-	env, err := newEnv()
-	if err != nil {
-		return
-	}
-
-	if readonly {
-		err = env.Open(path, lmdb.Readonly, 0440)
-	} else {
-		err = env.Open(path, 0, 0660)
-	}
-
-	if err != nil {
-		err = fmt.Errorf("failed to open the database in the "+
-			"directory %s: %s", path, err)
-		closeErr := env.Close()
-		if closeErr != nil {
-			err = fmt.Errorf("%s; failed to close the database: "+
-				"%s", err, closeErr)
-		}
-		return
-	}
-
-	d = &Database{
-		Path:     path,
-		env:      env,
-		Readonly: readonly}
-
-	return
-}
-
 // GetChannel returns the channel associated with the descriptor in the
 // database, if it exists; nil otherwise.
+//
+// GetChannel ensures:
+// * !(err == nil && channel != nil) || t.mustCount() > 0
+// * err == nil || channel == nil || channel.Descriptor_ == descriptor
+// * empty response for empty database: err == nil || t.mustCount() > 0 || channel == nil
```

Comment:
I'd have the mustCount contracts only in testing, since you don't want to run Count() in production.

I think you can rewrite the two contracts involving count just as:
```go
* empty response for empty database: !InTest || !(err == nil && t.mustCount == 0) || channel == nil
```

In other words, you assume that mustCount is correctly implemented, and then involve it in the contract here.